api4/channel_test.go

```diff
@@ -1162,18 +1030,14 @@ func TestDeleteChannel(t *testing.T) {
 	pass, resp := Client.DeleteChannel(publicChannel1.Id)
 	CheckNoError(t, resp)
 
-	if !pass {
-		t.Fatal("should have passed")
-	}
+	require.True(t, pass, "should have passed")
 
-	if ch, err := th.App.GetChannel(publicChannel1.Id); err == nil && ch.DeleteAt == 0 {
-		t.Fatal("should have failed to get deleted channel")
-	}
+	ch, err := th.App.GetChannel(publicChannel1.Id)
+	require.True(t, err != nil || ch.DeleteAt != 0, "should have failed to get deleted channel, or returned one with a populated DeleteAt.")
```

Comment:
can we break this into `require.NotNil(t, err)` and the other test `require.Zero(t, ...)`

Reply:
I may be mis-reading, but the original logic only fails if the error is both nil _and_ the returned value's `DeleteAt` time is still the zero value.  Hence the two conditionals are linked and can not be separated.  This is expressed by the test comment, which stats that either an error should have been generated or a non-zero `ch.DeleteAt` returned.